### ISSUE TITLE
TestSTMAbort: bad value

### ISSUE THREAD

**Issue by heyitsanthony** [coreos/etcd#4884]
```
--- FAIL: TestSTMAbort (0.09s)
	v3_stm_test.go:138: bad value. got &{Header:cluster_id:3234178572734379319 member_id:10344793263254342075 revision:2 raft_term:2  Kvs:[key:"foo" create_revision:2 mod_revision:2 version:1 value:"bap" ] More:false}, expected nothing
```

I think this is a bug in grpc; it's sending despite a canceled context.

**Comment by xiang90** [comment]
@heyitsanthony Are we able to reproduce the gRPC issue with a simple program?

**Comment by heyitsanthony** [comment]
@xiang90 there's not much happening with the STM code that would implicate it. I believe it's related to  https://github.com/grpc/grpc-go/blob/master/transport/http2_client.go#L236-L242 (it should be checking the cancelation status) but when I tried writing a test case against the latest grpc it didn't seem to trigger. I haven't tried to repro with the latest grpc + etcd yet.

The easy way to reproduce this is
```
while [ 1 ]; do go test --races -v -run=STM -cpu=`seq 1 32 | xargs echo | sed 's/ /,/g'` || break; done
```

**Comment by heyitsanthony** [comment]
Can reproduce on plain go-grpc. PR at https://github.com/grpc/grpc-go/pull/634

**Comment by xiang90** [comment]
Closed by #5061
